packages/common-all/src/BacklinkUtils.ts

```diff
@@ -0,0 +1,59 @@
+import _ from "lodash";
+import { DLink, NotePropsMeta } from "./types";
+
+export class BacklinkUtils {
+  /**
+   * Create backlink out of link if it references another note (denoted by presence of link.to field)
+   *
+   * @param link Original link to create backlink out of
+   * @returns backlink or none if not applicable
+   */
+  static createFromDLink(
+    link: DLink
+  ): (Omit<DLink, "type"> & { type: "backlink" }) | undefined {
+    const maybeToNoteFname = link.to?.fname;
+    if (maybeToNoteFname) {
+      return {
+        from: link.from,
+        type: "backlink",
+        position: link.position,
+        value: link.value,
+      };
+    }
+    return;
+  }
+
+  /** Adds a backlink by mutating the 'note' argument in place.
+   *
+   *  @param note note that the link is pointing to. (mutated)
+   *  @param link backlink to add. */
+  static addBacklink({
+    note,
+    backlink,
+  }: {
+    note: NotePropsMeta;
+    backlink: Omit<DLink, "type"> & { type: "backlink" };
+  }): void {
+    note.links.push(backlink);
```

Comment:
any chance we have duplicates?

Reply:
Definitely could happen. Do you think its worth the extra runtime to perform a dup check?

Reply:
I think it's worth doing the check (shouldn't be too expensive).  Otherwise we may have funky UI in the backlinks panel and potentially in other places as well.